lib/cc/config/prepare.rb

```diff
@@ -0,0 +1,68 @@
+module CC
+  module Config
+    class Prepare
+      class NoPrepareNeeded
+        def fetch
+          []
+        end
+      end
+
+      attr_reader :fetch
+
+      def self.from_yaml(data)
+        if data.present?
+          fetch = Fetch.from_yaml(data.fetch("fetch", []))
+
+          new(fetch: fetch)
+        else
+          NoPrepareNeeded.new
+        end
+      end
+
+      def initialize(fetch:)
+        @fetch = fetch
+      end
+
+      class Fetch
+        def self.from_yaml(data)
```

Comment:
My current approach is to write all of this consuming code with the assumption that the config has been validated. We'll maintain `codeclimate validate-config` (and run that on as a step on hosted still); it will go through `cc-yaml` *for now*, but be encapsulated in a class that we can rewrite later (see #638)